lib/shared/bucketing-assembly-script/assembly/eventQueue/requestPayloadManager.ts

```diff
@@ -0,0 +1,173 @@
+            const variableFeatureVarAggMapKeys = variableFeatureVarAggMap.keys()
+
+            for (let y = 0; y < variableFeatureVarAggMapKeys.length; y++) {
+                const variableKey = variableFeatureVarAggMapKeys[y]
+                const featureVarAggMap: Map<string, Map<string, i64>> = variableFeatureVarAggMap.get(variableKey)
+
+                let value: f64 = NaN
+                if (featureVarAggMap.has('value')) {
+                    const varAggMap = featureVarAggMap.get('value')
+                    if (varAggMap.has('value')) {
+                        value = f64(varAggMap.get('value'))
+
+                        // Add aggVariableDefaulted Events
+                        const dvcEvent = new DVCEvent(type, variableKey, null, value, null)
+                        aggEvents.push(new DVCRequestEvent(dvcEvent, user_id, null))
+                    } else {
+                        throw new Error('Missing sub value map to write aggVariableDefaulted events')
+                    }
+                } else {
+                    const featureVarAggMapKeys = featureVarAggMap.keys()
+
+                    for (let x = 0; x < featureVarAggMapKeys.length; x++) {
+                        const _feature = featureVarAggMapKeys[x]
+                        const variationAggMap: Map<string, i64> = featureVarAggMap.get(_feature)
+                        const variationAggMapKeys = variationAggMap.keys()
+
+                        for (let z = 0; z < variationAggMapKeys.length; z++) {
+                            const _variation = variationAggMapKeys[z]
+                            value = f64(variationAggMap.get(_variation))
+
+                            const metaData = new JSON.Obj()
+                            metaData.set('_feature', _feature)
+                            metaData.set('_variation', _variation)
+
+                            // Add aggVariableEvaluated Events
+                            const dvcEvent = new DVCEvent(type, variableKey, null, value, metaData)
+                            aggEvents.push(new DVCRequestEvent(dvcEvent, user_id, null))
+                        }
+                    }
+                }
+            }
+        }
+
+        // Generate defaulted agreagate user as our events APIs require a user / user_id
+        const dvcUser = new DVCPopulatedUser(new DVCUser(
+            user_id, null, null, null, null, NaN, null, null, null, null
+        ))
+        this.addEventsToPendingPayloads(new UserEventsBatchRecord(dvcUser, aggEvents))
+    }
+
+    /**
+     * Chunk up UserEventsBatchRecord's into payloads of size: this.chunkSize
+     */
+    private addEventsToPendingPayloads(record: UserEventsBatchRecord): void {
+        let flushPayload = new FlushPayload([])
+
+        while (record.events.length > this.chunkSize) {
+            const batchRecord = new UserEventsBatchRecord(
+                record.user,
+                record.events.splice(0, this.chunkSize)
```

Comment:
it seems a bit weird to directly mutate the records that are being passed in here